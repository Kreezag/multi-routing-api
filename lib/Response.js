import { isNumber } from './utils';


export default class Response {
  constructor ({ ok, status, result, rawResponse = null, options = {} }) {
    if (process.env.NODE_ENV !== 'production') {
      if (result === undefined) {
        throw new Error('result is undefined');
      }

      if (ok !== false && ok !== true) {
        throw new Error('ok is not a boolean');
      }

      if (!isNumber(status)) {
        throw new Error('status is not a number');
      }
    }

    this.ok = Boolean(ok);
    this.status = Number(status);
    this.result = result;
    this.options = options;
    this.rawResponse = rawResponse;
  }
}
