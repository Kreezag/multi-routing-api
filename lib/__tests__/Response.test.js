/* eslint-env browser, jest */

import Response from '../Response';


describe('Response', () => {
  it('need to throw if passed invalid params', () => {
    expect(() => {
      const transport = new Response();
    }).toThrow();

    expect(() => {
      const transport = new Response({});
    }).toThrow();

    expect(() => {
      const transport = new Response({ result: null });
    }).toThrow();

    expect(() => {
      const transport = new Response({ ok: false, result: null });
    }).toThrow();
  });

  it('should set needable props', () => {
    const transport = new Response({ ok: false, result: null, status: 404 });

    expect(transport).toMatchSnapshot();
  });
});
