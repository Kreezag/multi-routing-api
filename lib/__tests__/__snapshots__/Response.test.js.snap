exports[`Response should set needable props 1`] = `
Response {
  "ok": false,
  "options": Object {},
  "rawResponse": null,
  "result": null,
  "status": 404,
}
`;
